python/third_loader.py

```diff
@@ -0,0 +1,106 @@
+import datetime
+import re
+from more_itertools import first
+import requests
+from bs4 import BeautifulSoup
+import sys
+from modules.utils import auth_token_generator, curve_file_generator
+
+CZC_Table_Map = [
+    'date',
+    'hour',
+    'bg,price,euro/mwh',
+    'gr,price,euro/mwh',
+    'bg,gr,atc,euro/mwh',
+    'gr,bg,atc,euro/mwh',
+    'bg,gr,crossborderflow,euro/mwh',
+    'gr,bg,crossborderflow,euro/mwh'
+]
+
+BGRO_CZC_Table_Map = [
+    'date',
+    'hour',
+    'bg,price,euro/mwh',
+    'ro,price,euro/mwh',
+    'bg,ro,atc,euro/mwh',
+    'ro,bg,atc,euro/mwh',
+    'bg,ro,crossborderflow,euro/mwh',
+    'ro,bg,crossborderflow,euro/mwh'
+]
+
+
+def check_if_data_available_for(start_date: datetime.date):
+    if start_date < datetime.date(year=2024, month=1, day=24):
+        raise ValueError(f"{start_date} is prior to the data publish start date of January 24th, 2024.")
+    if start_date > datetime.datetime.now().date() + datetime.timedelta(days=1):
+        raise ValueError(f"{start_date} is in the future. Query date is supported till today's date.")
+
+def map_table_data(table: list, Table_Map: list):
+    table_data = []
+    for row in table[1:]:
```

Comment:
Reason behind first row skipped [Should be explicitly mentioned]

Reply:
The first row contains the key names (ie, date, hour, bg_gr_czc etc).Will mention with a comment here

Reply:
Ok then the first row represent the table header.
This can be used to match with our defined header to verify source data consistency and avoid wrong data publish